HMGB1.py

```diff
@@ -0,0 +1,422 @@
+from Bio import SeqIO
+from abc import ABC, abstractclassmethod
+from Bio.SeqUtils import GC123
+
+
+RETRANSLATION_DICT = {
+        'F': 'TTC', 'f': 'ttc',
+        'L': 'TTA', 'l': 'tta',
+        'S': 'TCG', 's': 'tcg',
+        'Y': 'TAC', 'y': 'tac',
+        'C': 'TGC', 'c': 'tgc',
+        'W': 'TGG', 'w': 'tgg',
+        'P': 'CCC', 'p': 'ccc',
+        'H': 'CAT', 'h': 'cat',
+        'Q': 'GAA', 'q': 'gaa',
+        'R': 'CGA', 'r': 'cga',
+        'I': 'ATT', 'i': 'att',
+        'M': 'ATG', 'm': 'atg',
+        'T': 'ACC', 't': 'acc',
+        'N': 'AAT', 'n': 'aat',
+        'K': 'AAA', 'k': 'aaa',
+        'V': 'GTT', 'v': 'gtt',
+        'A': 'GCA', 'a': 'gca',
+        'D': 'GAT', 'd': 'gca',
+        'E': 'GAG', 'e': 'gag',
+        'G': 'GGG', 'g': 'ggg'
+    }
+
+
+THREE_LETTER_ALPHABET = {
+                'A': 'ALA', 'R': 'ARG', 'N': 'ASN', 'D': "ASP", 'V': 'VAL',
+                'H': 'HIS', 'G': "GLY", 'Q': "GLN", 'E': 'GLU', 'I': 'ILE',
+                'L': 'LEU', 'K': 'LYS', 'M': 'MET', 'P': 'PRO', 'S': 'SER',
+                'Y': 'TYR', 'T': 'THR', 'W': 'TRP', 'F': 'PHE', 'C': 'CYS',
+                'a': 'ala', 'r': 'arg', 'n': 'asn', 'd': "asp", 'v': 'val',
+                'h': 'his', 'g': "gly", 'q': "gln", 'e': 'glu', 'i': 'ile',
+                'l': 'leu', 'k': 'lys', 'm': 'met', 'p': 'pro', 's': 'ser',
+                'y': 'tyr', 't': 'thr', 'w': 'trp', 'f': 'phe', 'c': 'cys'
+    }
+
+
+class BiologicalSequence(ABC):
+    @abstractclassmethod
+    def __len__():
+        pass
+
```

Comment:
я бы еще добавил `__repr__`
```suggestion
    def __repr__(self):
        return f'NucleicAcid("{self.sequence}")'
```